cmake/initialize-submodules.cmake

```diff
@@ -1,27 +1,89 @@
-macro(initialize_submodules)
+#update the named submodule
+
+#attempt to add the submodule
+#optional arguments:
+# 1 - path
+# 2 - alternate reference name
+macro(add_submodule module_name)
+
+  message(STATUS "Attempting to add \"${module_name}\"")
```

Comment:
Is this a remnant from debugging? I think we can remove all but maybe one or two of the STATUS messages. Either it was successful or it is a FATAL_ERROR.